packages/vscode-extension/src/debug/commonUtils.ts

```diff
@@ -143,4 +144,34 @@ async function getLocalDebugConfig(key: string): Promise<string | undefined> {
 
 export async function getSkipNgrokConfig(): Promise<string | undefined> {
   return getLocalDebugConfig(constants.skipNgrokConfigKey);
+}
+
+async function detectPortListeningImpl(port: number, host: string): Promise<boolean> {
+  return new Promise<boolean>((resolve, reject) => {
+    try {
+      const server = net.createServer();
+      server.once("error", (err) => {
+              if (err.message.includes("EADDRINUSE")) {
+                resolve(true);
+              }
+            })
+            .once("listening", () => {
+              server.close();
```

Comment:
Does `server.close()` release the port immediately? Or, need some timeout?

Reply:
Practically, `server.close()` will release the port immediately since there is no connection established on this port.

Reply:
Added `once("close")` to solve this potential problem:
```
            .once("listening", () => {
              server.close();
            })
            .once("close", () => {
              resolve(false);
            })
```